cmd/linkcheck/links.go

```diff
@@ -14,39 +14,47 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-//This command checks if the hyperlinks in files are valid. It checks the files
-//with 'fileSuffix' in 'rootDir' for URLs that match 'prefix'. It trims the
-//'prefix' from the URL, uses what's left as the relative path to repoRoot to
-//verify if the link is valid. For example:
-//$ linkcheck --root-dir=${TYPEROOT} --repo-root=${KUBE_ROOT} \
-//  --file-suffix=types.go --prefix=http://releases.k8s.io/HEAD
+// This tool extracts the links from types.go and .md files, visits the link and
+// check the status code of the response.
+// Usage:
+// $ linkcheck --root-dir=${ROOT}
 
 package main
 
 import (
 	"fmt"
 	"io/ioutil"
+	"net/http"
 	"os"
-	"path"
 	"path/filepath"
 	"regexp"
-	"strings"
+	"time"
 
+	"github.com/mvdan/xurls"
 	flag "github.com/spf13/pflag"
 )
 
 var (
-	httpRE *regexp.Regexp
-
 	rootDir    = flag.String("root-dir", "", "Root directory containing documents to be processed.")
-	repoRoot   = flag.String("repo-root", "", `Root directory of k8s repository.`)
-	fileSuffix = flag.String("file-suffix", "", "suffix of files to be checked")
-	prefix     = flag.String("prefix", "", "Longest common prefix of the link URL, e.g., http://release.k8s.io/HEAD/ for links in pkg/api/types.go")
+	fileSuffix = flag.StringSlice("file-suffix", []string{"types.go", ".md"}, "suffix of files to be checked")
+	whiteList  = []*regexp.Regexp{
+		regexp.MustCompile(`https://kubernetes-site\.appspot\.com`),
+		regexp.MustCompile(`http://www\.apache\.org/licenses/LICENSE-2\.0`),
+		regexp.MustCompile(`http://kubernetes\.io/img/warning\.png`),
+		regexp.MustCompile(`http://releases\.k8s\.io$`),
+		// skip url that doesn't start with an English alphabet.
+		regexp.MustCompile(`https?://[^A-Za-z].*`),
+		regexp.MustCompile(`https?://localhost.*`),
+	}
+	htmlpreviewReg = regexp.MustCompile(`https://htmlpreview\.github\.io/\?`)
+	httpOrhttpsReg = regexp.MustCompile(`https?.*`)
 )
 
-func newWalkFunc(invalidLink *bool) filepath.WalkFunc {
+func newWalkFunc(invalidLink *bool, fileSuffixTable map[string]struct{}, client *http.Client) filepath.WalkFunc {
 	return func(filePath string, info os.FileInfo, err error) error {
-		if !strings.HasSuffix(info.Name(), *fileSuffix) {
+		_, found1 := fileSuffixTable[filepath.Ext(filePath)]
```

Comment:
Its not clear what are you trying to find here.
Maybe use better variable name than found1 and found2?